src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal static class TypeImportCompletionItem
+    {
+        private const string GenericTypeNameManglingString = "`";
+        private static readonly string[] s_IntegerOneToNine = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+        private static readonly string[] s_aritySuffixesOneToNine = { "`1", "`2", "`3", "`4", "`5", "`6", "`7", "`8", "`9" };
+
+        private const string ContainingNamespaceName = nameof(ContainingNamespaceName);
+        private const string TypeArityName = nameof(TypeArityName);
+
+        public static CompletionItem Create(INamedTypeSymbol typeSymbol, string containingNamespace)
+        {
+            var builder = PooledDictionary<string, string>.GetInstance();
+            builder.Add(ContainingNamespaceName, containingNamespace);
+
+            if (typeSymbol.Arity > 0)
+            {
+                builder.Add(TypeArityName, GetInteger(typeSymbol.Arity));
+            }
+
+            // TODO: Suffix should be language specific, i.e. `(Of ...)` if triggered from VB.
```

Comment:
def an issue :)

Reply:
Will fix later™

Reply:
We are, but I have been mostly focusing on C# and VB side might not be well thought out. Given this is an experimental feature, it's probably OK worst case we just disable it for VB for now.

Reply:
or just hack a check of `if c# "<>" else "(Of ...)"` :)

Reply:
@CyrusNajmabadi  well, I guess that'd work.